src/cargo/util/config/mod.rs

```diff
@@ -211,6 +213,16 @@ impl Config {
             })
             .collect();
 
+        let mut upper_case_env: HashMap<String, String> = HashMap::new();
+
+        if !cfg!(windows) {
+            upper_case_env = env
+                .clone()
+                .into_iter()
+                .map(|(k, _)| (k.to_uppercase().replace("-", "_"), k))
+                .collect();
+        }
```

Comment:
This is just a minor stylistic choice, but I think it is a little clearer to initialize only once, something like this:

```rust
        let upper_case_env = if cfg!(windows) {
            HashMap::new()
        } else {
            env.clone()
                .into_iter()
                .map(|(k, _)| (k.to_uppercase().replace("-", "_"), k))
                .collect()
        };
```